.circleci/config.yml

```diff
@@ -1,51 +1,130 @@
-version: 2
+version: 2.1 # Set the CI version.
 
+
+# We use orbs to provide some common ruby and node tasks and shorten our config.
+# Learn more about orbs: https://circleci.com/orbs/
+orbs:
+  ruby: circleci/ruby@0.1.2
+  node: circleci/node@1.1.6
+
+
+# Yaml References enable us to DRY our our config by sharing variables across multiple jobs.
+# In this case, we are commonly using the "workspaces" feature to share
+# build artifacts and files across jobs. For example, we build our Javascript and
+# persist it to a workspace to be made available when the Jekyll site builds.
 references:
-  js_deps_cache_key: &js_deps_cache_key
-    v8-dependency-js-deps-{{ checksum "package-lock.json" }}
-  js_deps_backup_cache_key: &js_deps_backup_cache_key
-    v8-dependency-js-deps
   workspace_root: &workspace_root
     /tmp/workspace
   attach_workspace: &attach_workspace
     attach_workspace:
       at: *workspace_root
 
-jobs:
-  js_build:
-    docker:
-      - image: node:8.11.1
+
+# Several steps in this config use the same, specialized ruby-caching steps.
+# Commands can be used to extract a common set of steps into a reusable-block.
+# In this case, whenever you see `- ruby-deps` as a "step" in the config below,
+# It is executing a command to restore_cache, install deps if needed, and save_cache.
+commands:
+  ruby-deps:
+    description: "Runs specialized Ruby cache steps."
     steps:
-      - *attach_workspace
-      - checkout
       - restore_cache:
-          keys:
-            - *js_deps_cache_key
-            - *js_deps_backup_cache_key
-      - run:
-          name: npm dependencies
-          command: npm install
+          key: circleci-docs-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
       - run:
-          name: Prepare JS assets
+          name: Install Ruby dependencies
+          command: bundle check --path=vendor/bundle || bundle install --path=vendor/bundle --jobs=4 --retry=3
+      - save_cache:
+          key: circleci-docs-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
+          paths:
+            - "vendor/bundle"
+
+
+jobs:
+  js_build: # this job is responsible for building Javascript assets and making them available for the "build" job
+    executor: # we specify our executor to use the node orb.
```

Comment:
I'm not clear on the syntax here: looking at https://circleci.com/docs/2.0/configuration-reference/#executors-requires-version-21 but can't see how you're defining executors documented?